src/file.ts

```diff
@@ -22,10 +28,16 @@ if (url.searchParams.has("from-file") && "launchQueue" in window && "files" in L
   dialog.showModal();
 }
 
+if (!("showOpenFilePicker" in window) || !("showSaveFilePicker" in window)) {
+  openButton.disabled = true;
```

Comment:
It might be better to avoid opening the modal in the first place rather than opening it and disabling the buttons.  We could even open a different modal with a link to a website that lists which browsers support the feature instead of using the default alert that can be dismissed by the user and doesn't align with our website style.